examples/1_Spectra_handling/custom_Lorentzian_broadening.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+"""
+========================
+Using a custom broadening function for atomic spectra
+========================
+
+The `lbfunc` parameter of :py:class:`~radis.lbl.factory.SpectrumFactory` allows us to specify a custom function to use for the Lorentzian broadening of a spectrum. This is especially useful for handling pressure broadening of atomic lines for which multiple theories exist.
+
+By default, RADIS calculates the total Lorentzian broadening of atomic lines as the sum of the Van der Waals broadening, Stark broadening, and radiation broadening as returned by :py:func:`~radis.lbl.broadening.gamma_vald3` from the broadening parameters for each of these types provided in the databank. The line shift is also calculated as 2/3 times the Van der Waals broadening, and `lbfunc` allows you to specify an alternative line shift as the 2nd return argument if you wish.
+
+`lbfunc` can be changed on the fly by changing the `sf.params.lbfunc` attribute of the :py:class:`~radis.lbl.factory.SpectrumFactory` instance, the result of which is reflected next time the library calculates its Lorentzian broadening.
+
+To make use of it, familiarise yourself with the column names that RADIS assigns to the relevant quantities you intend to use in `lbfunc`.
+
+We take an example of neutral atomic Oxygen and calculate a spectrum using the broadening formula (and default values) of SpectraPlot (https://spectraplot.com/), and assume there is no line shift.
+"""
+from radis import calc_spectrum
+
+def lbfunc1(**kwargs):
+    return 0.1*(296/kwargs['Tgas'])**0.8, None
+
+mole_fraction = 0.01
+
+s1, sf = calc_spectrum(
+    12850,
+    12870,
+    species="O_I",
+    Tgas=4000,
+    databank="kurucz",
+    pressure=1.01325, # = 1 atm
+    diluent={'H':1-mole_fraction-1e-3, 'e-': 1e-3}, #so it all adds up to 1
+    mole_fraction=mole_fraction,
+    path_length=15,
+    lbfunc=lbfunc1,
+    return_factory=True,
+    save_memory=False #to be able to recalculate spectra with the same SpectrumFactory
+)
+
+#%%
+# Now compare the result with that of the default handling of broadening in RADIS by removing the `lbfunc` parameter, recalculating the spectrum without it, and plotting the diff between the results:
+#
+sf.params.lbfunc = None
+s_default = sf.eq_spectrum(4000)
+
+from radis import plot_diff
+
+plot_diff(s1, s_default, label1='s1', label2='s_default')
```

Comment:
I would prefer to directly use `SpectrumFactory` here. Again, we both know it's possible to use calc_spectrum and kwargs for that, but let's keep it simple for calc_spectrum, and more advanced for SpectrumFactory.

- [ ] Use only SpectrumFactory
- [ ] Start with the default parameters `s1 = sf.eq_spectrum(Tgas = 4000, name = "RADIS default - Kurucz parameters")`
- [ ] Then compute with the custom lbfun, `s2 = sf.eq_spectrum(Tgas = 4000, name = "SpectraPlot default")`
- [ ] Compare as you did